### PR TITLE
Lilith101 keyboard addition

### PR DESCRIPTION
**Pull request by St0RM53**
## Description

- Added new keyboard: lilith101

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lilith101/custom_keys.c
Comment (zvecr) [inline keyboards/lilith101/custom_keys.c:1]: As this file implements `_user` functions, including keymap specific keycodes, it should not exist at the keyboard level.

It should be moved to `keyboards/lilith101/keymaps/default/custom_keys.c` and a `keyboards/lilith101/keymaps/default/rules.mk` added to set `SRC += custom_keys.c` at the keymap level.

File: keyboards/lilith101/info.json [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        "oled": true
+
+    },
+    "matrix_pins": {
+        "cols": ["B16", "B17", "D0", "A12", "A13", "D7", "D4", "D2", "D3", "C3", "C4", "C6", "C7", "D1", "C0", "B0", "B1", "B3", "B2"],
+        "rows": ["D5", "D6", "C1", "C2", "A5", "B19"]
+    },
+    "encoder": {
+        "rotary": [
+            { "pin_a": "B18", "pin_b": "A4", "resolution": 4 } /* pin_b on 1st Prototype is on C5, 2nd Prototype on A4 */
```
Comment (zvecr) [inline keyboards/lilith101/info.json:24]:
```suggestion
            { "pin_a": "B18", "pin_b": "A4" } /* pin_b on 1st Prototype is on C5, 2nd Prototype on A4 */
```

File: keyboards/lilith101/info.json [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        "vid": "0xFEED"
+    },
+    "indicators": {
+        "num_lock": "C9",
+        "caps_lock": "C8",
+        "scroll_lock": "E0",
+        "on_state": 1
+    },
+    "rgb_matrix": {
+        "layout": [
```
Comment (zvecr) [inline keyboards/lilith101/info.json:41]:
```suggestion
        "driver": "ws2812",
        "layout": [
```

File: keyboards/lilith101/rules.mk [hunk 3]
```diff
@@ -0,0 +1,7 @@
+SRC += custom_keys.c
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+#WS2812_DRIVER = bitbang #Default value
```
Comment (zvecr) [inline keyboards/lilith101/rules.mk:4]:
```suggestion
```

File: keyboards/lilith101/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# lilith101
+
+![lilith101](imgur.com image replace me!)
+
+*A short description of the keyboard/project*
```
Comment (zvecr) [inline keyboards/lilith101/readme.md:5]: Should be filled in

File: keyboards/lilith101/info.json [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true,
+        "oled": true
+
```
Comment (zvecr) [inline keyboards/lilith101/info.json:16]:
```suggestion
        "oled": true,
        "rgb_matrix": true
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/lilith101/custom_keys.c
Comment (St0RM53) [reply keyboards/lilith101/custom_keys.c:1]: I am getting implicit declaration of function errors for all `rgb_matrix_function()` functions after i moved it :/

File: keyboards/lilith101/info.json (hunk 2 above)
Comment (St0RM53) [reply keyboards/lilith101/info.json:41]: Needs to be capitalized: `WS2812`

**Pull request closed without merging**